### PR TITLE
DOC: fix broken links to modules

### PR DESCRIPTION
**Pull request by partev**
fix 3 broken links to modules: random, ctypes, and decimal.

https://docs.python.org/3.12/license.html#mersenne-twister

https://docs.python.org/3.12/license.html#libffi

https://docs.python.org/3.12/license.html#libmpdec

### CODE REVIEW

**Comment by slateny** [comment]
By modules, I believe it refers to the C modules under [`Modules`](https://github.com/python/cpython/tree/main/Modules), not the Python ones under [`Lib`](https://github.com/python/cpython/tree/main/Lib). For example, the comments in [_randommodule.c](https://github.com/python/cpython/blob/main/Modules/_randommodule.c) match up with what's listed in the license file.

**Comment by partev** [comment]
> By modules, I believe it refers to the C modules under [`Modules`](https://github.com/python/cpython/tree/main/Modules), not the Python ones under [`Lib`](https://github.com/python/cpython/tree/main/Lib). For example, the comments in [_randommodule.c](https://github.com/python/cpython/blob/main/Modules/_randommodule.c) match up with what's listed in the license file.

sounds plausible. Should then :mod: keyword be removed then?

**Comment by slateny** [comment]
Actually a correction - `ctypes` doesn't have a module under `Lib` so the change would make sense. As for keeping or changing `_random`/`_decimal`, I'm not too confident on that part but I feel that it should just be kept as it is now.

@CAM-Gerlach Could you double check my thoughts here?

File: Doc/license.rst [hunk 1]
```diff
@@ -302,7 +302,7 @@ for third-party software incorporated in the Python distribution.
 Mersenne Twister
 ----------------
 
-The :mod:`_random` module includes code based on a download from
+The :mod:`random` module includes code based on a download from
```
Comment (CAM-Gerlach) [inline Doc/license.rst]:
```suggestion
The :mod:`!_random` C extension underlying the :mod:`random` module
includes code based on a download from
```

File: Doc/license.rst [hunk 2]
```diff
@@ -819,7 +819,7 @@ sources unless the build is configured ``--with-system-expat``::
 libffi
 ------
 
-The :mod:`_ctypes` extension is built using an included copy of the libffi
+The :mod:`ctypes` extension is built using an included copy of the libffi
```
Comment (CAM-Gerlach) [inline Doc/license.rst]:
```suggestion
The :mod:`!_ctypes` C extension underlying the :mod:`ctypes` module
is built using an included copy of the libffi
```

File: Doc/license.rst [hunk 3]
```diff
@@ -920,7 +920,7 @@ on the cfuhash project::
 libmpdec
 --------
 
-The :mod:`_decimal` module is built using an included copy of the libmpdec
+The :mod:`decimal` module is built using an included copy of the libmpdec
```
Comment (CAM-Gerlach) [inline Doc/license.rst]:
```suggestion
The :mod:`!_decimal` C extension underlying the :mod:`decimal` module
is built using an included copy of the libmpdec
```

**Review by CAM-Gerlach — COMMENTED** [review]
Sorry for the delay, You can prefix a reference name with `!` to not resolve the reference, and just format the contents with the appropriate styling, i.e. `` :mod:`!_random` `` to mention the extension module, formatted as a module but without trying to resolve the reference, and you can also use a different display name as the actual resolved reference, i.e. `` :mod:`_random <random>` to specifically refer to the `_random` extension module while referring to the overall documentation for the `random` module.

However, given the evident confusion here over the ambiguity between a Python "module" and the C extension modules in the CPython `Modules` directory, and the need for clarity as far as licensing is concerned, I suggest specifically separating the two, as my GitHub suggestions illustrate, which link to the name/documentation for the overall module, but also make clear that the specific code the license refers to is the `_`-prefixed C extension module component.

I played around with several other words to describe the relationship other than "underlying", including "supporting", "for", "of", etc. and am not sure what people would like best here, but left it for now.

**Comment by CAM-Gerlach** [comment]
> Actually a correction - ctypes doesn't have a module under Lib so the change would make sense. As for keeping or changing _random/_decimal, I'm not too confident on that part but I feel that it should just be kept as it is now.

It actually has a whole package under `Lib`.

**Pull request merged**